progfun/tutoriales/Haskell/Haskell Language_files/tryhaskell.pages.js

```diff
@@ -0,0 +1,459 @@
+            }
+        },
+        {
+            guide: function (result) {
+                if (!result || !result.value) result = { value: "[1,4,9,16,25,36,49,64,81,100]" };
+                return "<h3>Déjalas ser funciones</h3>" +
+
+                    "<p> Acabaste de escribir una función llamada <code>square</code> y luego " +
+                    "le añadiste otra función (<code>map</code>) y obtuviste<code>" +
+                    tryhaskell.pages.htmlEncode(result.value) + "</code>, que era lo que esperabas!</p>" +
+
+                    "<p>Haskell es experto en componer y unas de las funciones " +
+                    "que podemos realizar son:</p>" +
+
+                    "<ul>" +
+                    "<li><code>let add1 x = x + 1 in map add1 [1,5,7]</code></li>" +
+                    "<li><code>let take5s = filter (==5) in take5s [1,5,2,5,3,5]</code></li>" +
+                    "<li><code>let take5s = filter (==5) in map take5s [[1,5],[5],[1,1]]</code></li>" +
+                    "</ul>" +
+
+                    "<p>Obtuviste lo que esperabas?</p>" +
+
+                    "<p>Prueba ahora con un texto; Cómo lo vuelves a letras mayúsculas?</p>" +
+
+                    "<p><code>toUpper 'a'</code></p>"
+            },
+            trigger: function (result) {
+                return result.expr.match(/^[ ]*let[ ]+square[ ]+x[ ]*=[ ]*x[ ]*\*[ ]*x[ ]*in[ ]+map[ ]+square[ ]*\[1..10\][ ]*$/) &&
+                    (result.type.match(/^\(?Num [a-z], Enum [a-z]\)? => \[[a-z]\]$/) ||
+                        result.type.match(/^\(?Enum [a-z], Num [a-z]\)? => \[[a-z]\]$/));
+            }
+        },
+        {
+            guide: function (result) {
+                return "<h3>¿Te gustan los retos?</h3>" +
+
+                    "<p>Okayyy! Solo recuerda: los caracteres se escriben como <code>'a'</code> y " +
+                    "strings (lists of characters)se escriben como <code>\"a\"</code>." +
+
+                    "<p>Me gustaría que convirtieras<code>toUpper</code> mi nombre en mayúscula,por favor.</p> " +
+                    +"<p> Me llamo <code>\"Haskellito\"</code>.</p> " +
+                    "<p> Ahora hazlo! Creo en ti, sé el primero en resolverlo. Te conviene...</p>" 
+
+                    // '<p>Spoiler: <code class="spoiler">map toUpper "Chris"</code></p>'
+            },
+            trigger: function (result) {
+                return result.expr.match(/^toUpper 'a'$/) &&
+                    result.type == "Char";
+            }
+
+            //DEJAR HASTA ACÁ!! POR TIEMPPOOOOOOO
+        },
+        {
+            guide: function (result) {
+                return "<h3>Lo Lograsteeeee, crack!</h3>" +
+
+                    "<p>Avísale a alguno de los expositores " +
+                    "para recibir tu premio.</p>" +
+
+                    "<p>Espero te haya gustado ese lenguaje."+
+                    " Te recomendamos aprenderlo a fondo si quieres aprender a programar mucho mejor.</p>"
```

Comment:
```suggestion
                    " Te recomendamos estudiarlo a fondo si quieres aprender a programar mucho mejor.</p>"
```
Eliminar redundancia/repetición.